### PR TITLE
feat: Convert benchmark.sh to python and add unit tests

### PR DESCRIPTION
**Pull request by ejlee3**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/performance-tools/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**issue_number**

## Test Instructions if applicable

Prerequisites:
- python environment with Python 3.12.2 (can use [miniconda](https://rsd-intel.slack.com/archives/C02E1TXQS8Z/p1711055235807679?thread_ts=1711050322.426669&cid=C02E1TXQS8Z) and create a [Python 3.12.2 env](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-python.html))
- installed necessary packages listed in requirements.txt

1. Pull the latest changes from this branch and from the retail-use-cases for updated compose files
2. change directories to the `benchmark-script` directory
3.  change directories `cd docker` and run `make build-igt` to build the benchmark docker container
4. change directories back `cd ..`
5. Ensure that the benchmarking script works run something like this
    ```
    python benchmark.py --compose_file ../../retail-use-cases/use-cases/gst_capi/add_camera-simulator.yml --compose_file ../../retail-use-cases/use-cases/gst_capi/add_gst_capi_yolov5_ensemble.yml
    ```
    **Note**: it is possible to provide other input variables - use `python benchmark.py --help` to see all available args
6.  run the unit tests in benchmark_test.py
    ```
    python -m unittest -v benchmark_test.py
    ```

### CODE REVIEW

File: benchmark-scripts/benchmark.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                              compose_post_args="",
+                              env_vars=os.environ.copy()):
+    try:
+        files = " -f ".join(compose_files)
+        compose_string = ("docker compose %s -f %s %s %s" %
+                          (compose_pre_args, files, command,
+                           compose_post_args))
+        compose_args = shlex.split(compose_string)
+
+        p = subprocess.Popen(compose_args, stdout=subprocess.PIPE,
```
Comment (jim-wang-intel) [inline benchmark-scripts/benchmark.py:59]: add # nosec to ignore for now

File: benchmark-scripts/benchmark_test.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+'''
+* Copyright (C) 2024 Intel Corporation.
+*
+* SPDX-License-Identifier: Apache-2.0
+'''
+
+import mock
+import subprocess
```
Comment (jim-wang-intel) [inline benchmark-scripts/benchmark_test.py:8]: add # nosec ignore for now

File: benchmark-scripts/benchmark_test.py
Comment (jim-wang-intel) [inline benchmark-scripts/benchmark_test.py:1]: let's add a make file target for running unit tests and get the code coverage please

File: benchmark-scripts/benchmark.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        os.mkdir(results_dir)
+
+    print("Starting workload(s)")
+
+    # start the docker containers
+    # pass in necessary variables using env vars
+    compose_files = []
+    for file in my_args.compose_file:
+        compose_files.append(os.path.abspath(file))
+    compose_files.append(os.path.abspath(os.path.join(
```
Comment (brian-intel) [inline benchmark-scripts/benchmark.py:88]: Can you add a comment to specify that this is the benchmark docker-compose file.

File: benchmark-scripts/Makefile [hunk 4]
```diff
@@ -17,4 +17,10 @@ run:
 	docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v `pwd`/results:/tmp/results --net=host --privileged benchmark:dev /bin/bash
 
 consolidate:
-	docker run -itd -v `pwd`/$(ROOT_DIRECTORY):/$(ROOT_DIRECTORY) -e ROOT_DIRECTORY=$(ROOT_DIRECTORY)--net=host --privileged benchmark:dev /bin/bash -c "python3 consolidate_multiple_run_of_metrics.py --root_directory $(ROOT_DIRECTORY)/ --output $(ROOT_DIRECTORY)/summary.csv"
\ No newline at end of file
+	docker run -itd -v `pwd`/$(ROOT_DIRECTORY):/$(ROOT_DIRECTORY) -e ROOT_DIRECTORY=$(ROOT_DIRECTORY)--net=host --privileged benchmark:dev /bin/bash -c "python3 consolidate_multiple_run_of_metrics.py --root_directory $(ROOT_DIRECTORY)/ --output $(ROOT_DIRECTORY)/summary.csv"
+
+python-test:
+	python -m unittest benchmark_test.py
```
Comment (jim-wang-intel) [inline benchmark-scripts/Makefile:23]: python coverage report doesn't work with this together yet, change to include the unit test report into coverage report:

```suggestion
	python -m coverage run -m unittest benchmark_test.py
```

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
makefile target `python-test` needs to change to include the coverage report data

File: benchmark-scripts/Makefile [hunk 5]
```diff
@@ -17,4 +17,10 @@ run:
 	docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v `pwd`/results:/tmp/results --net=host --privileged benchmark:dev /bin/bash
```
Comment (brian-intel) [inline benchmark-scripts/Makefile:17]: Add copyright header to Makefile

# Copyright © 2024 Intel Corporation. All rights reserved.
# SPDX-License-Identifier: Apache-2.0

**Review by brian-intel — CHANGES_REQUESTED** [review]

File: benchmark-scripts/camera-simulator.sh [hunk 6]
```diff
@@ -21,8 +21,6 @@ if [ "${COMMAND,,}" = "start" ]; then
     if [ -z "$CAMERAS" ]; then
```
Comment (brian-intel) [inline benchmark-scripts/camera-simulator.sh:21]: Copyright date

**Review by brian-intel — APPROVED** [review]
LGTM

**Review by jim-wang-intel — APPROVED** [review]
unit tests and coverage work

**Pull request merged**